src/modules/organizations/components/categories-editor.jsx

```diff
@@ -0,0 +1,106 @@
+import React from 'react';
+import DragReorderable from 'drag-reorderable';
+
+export default class CategoriesEditor extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.handleAddCategory = this.handleAddCategory.bind(this);
+    this.handleInputChange = this.handleInputChange.bind(this);
+    this.handleCategoryReorder = this.handleCategoryReorder.bind(this);
+    this.handleRemoveCategory = this.handleRemoveCategory.bind(this);
+    this.renderRow = this.renderRow.bind(this);
+    this.renderTable = this.renderTable.bind(this);
+  }
+
+  handleAddCategory() {
+    const newCategory = { key: Math.random(), category: 'Example' };
```

Comment:
I'm curious about why we use `Math.random()` to assign keys. Is there a reason why we can't use the item's position within the categories array as the key?

Reply:
great question. I originally went with `Math.random()` because I thought there may be issues when [using index as a key in React](https://zooniverse.slack.com/archives/C06LHUC6R/p1508401617000046). I tried using the label as key, but if one already exists called "Example" and I start trying to type "Example2" it disappears when I get to the word before the 2. 

I tried again using index, and this time setting it as the key similarly how I was with Math.random(), and finally treating the key as a true key (not an index, even though it may or may not be the right index), and I think it's properly working!